pkg/keymgmt/keymgmt.go

```diff
@@ -43,31 +44,36 @@ func GeneratePrivateKey(algorithm string) (interface{}, interface{}, error) {
 	default:
 		err = errors.New("Unsupported algorithm: " + algorithm)
 	}
-	
 	if err != nil {
 		return nil, nil, err
 	}
 
-	pub, err := getPublicKey(key)
+	pub, err := getPublicKeyBytes(key)
 	if err != nil {
 		return nil, nil, err
 	}
+
 	return key, pub, err
 }
 
-func getPublicKey(priv interface{}) (interface{}, error) {
+func getPublicKeyBytes(priv interface{}) ([]uint8, error) {
 	var err error
 	var pub interface{}
-
 	switch k := priv.(type) {
 	case *rsa.PrivateKey:
 		pub = &k.PublicKey
 	case *ecdsa.PrivateKey:
 		pub = &k.PublicKey
-	// The below would never happen (unless upstream crypto is broken, but lets log it anyway)
 	default:
 		err = errors.New("error generating public key" )
 	}
+	if err != nil {
+		return nil, err
+	}
```

Comment:
in some of the cosign code I actually gave up with this error and threw a panic (same as below during MarshalPKIXPublicKey). Like you said, an examination of the code shows no error can be thrown here. It really simplifies callers to not have to deal with this error. A panic seems appropriate. Your call though!

Reply:
I tend to agree with that, if the underlying code breaks there is really no recovering and its appropriate to panic.